packages/next/src/server/app-render/app-render.tsx

```diff
@@ -739,10 +739,7 @@ async function renderToHTMLOrFlightImpl(
       function onHeaders(headers: Headers): void {
         // Copy headers created by React into the response object.
         headers.forEach((value: string, key: string) => {
-          res.appendHeader(key, value)
```

Comment:
The headers here are actually assigned inside the `base-server.ts` during the render after the result is returned. This was previously throwing during revalidation causing React to hang.